src/main/java/org/spongepowered/common/mixin/optimization/world/gen/structure/MixinMapGenStructure_Structure_Saving.java

```diff
@@ -0,0 +1,30 @@
+package org.spongepowered.common.mixin.optimization.world.gen.structure;
+
+import net.minecraft.world.World;
+import net.minecraft.world.WorldSavedData;
+import net.minecraft.world.gen.MapGenBase;
+import net.minecraft.world.gen.structure.MapGenStructure;
+import net.minecraft.world.gen.structure.MapGenStructureData;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.common.interfaces.world.IMixinWorld;
+
+@Mixin(MapGenStructure.class)
+public abstract class MixinMapGenStructure_Structure_Saving extends MapGenBase {
+
+    @Redirect(method = "initializeStructureData", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;loadItemData(Ljava/lang/Class;Ljava/lang/String;)Lnet/minecraft/world/WorldSavedData;"))
+    private WorldSavedData checkLoadOrDefault(World world, Class<? extends WorldSavedData> structureClass, String structureName) {
+        if (world instanceof IMixinWorld) {
+            if (((IMixinWorld) world).getActiveConfig().getConfig().getStructures().structureMap.get(structureName)) {
```

Comment:
This will be `false` as far as I can see - the map has no values (other than `Mineshaft`, which itself is `false`) that are true, and is never populated.